facebookresearch/ParlAI
# CMU_DoG

**Patch description**
Adding CMU_DoG dataset. This is mostly a copypasta of code that wasn't yet checked in, and was originally written mostly by Moya and myself.

**Testing steps**
```
λ parlai dd -t cmu_dog
λ parlai dd -t cmu_dog --cmu-dog-split-type original
λ parlai dd -t cmu_dog --cmu-dog-split-type seen --datatype test
λ parlai dd -t cmu_dog --cmu-dog-split-type unseen --datatype test
λ parlai dd -t cmu_dog --cmu-dog-include-knowledge-keys movieName
λ parlai dd -t cmu_dog --cmu-dog-provide-movie-context False --mutators knowledge_only_when_updated,prepend_knowledge_to_message
λ parlai dd -t cmu_dog --cmu-dog-only-with-knowledge False --cmu-dog-rating 1 --cmu-dog-multi-msg-delimiter "__" --cmu-dog-fact-delimiter "/"
```
Unit tests
```
λ pytest -k test_zootasks
λ pytest --force-regen parlai/tasks/cmu_dog/test.py  # fails initially, as expected
λ pytest parlai/tasks/cmu_dog/test.py  # succeeds
```
